docs/madr/decisions/057-rework-mesh-routes-matching.md

```diff
@@ -0,0 +1,379 @@
+# Rework of Mesh*Routes matching
+
+* Status: accepted
+
+Technical story: https://github.com/kumahq/kuma/issues/10247
+
+## Context and Problem Statement
+
+At the moment when targeting Mesh*Route you need to put it in topLevel targetRef. 
+We want to iterate over policy design to fully utilize the fact that we are referencing real objects.
+The current implementation has made things a little tricky because policy matching is convoluted where the top level 
+MeshHTTPRoute gets converted to a fake `to`.
+
+## Considered options
+
+- add possibility to put Mesh*Routes in `spec.to[].targetRef`
+- leave Mesh*Routes in top level targetRef as it is now
```

Comment:
I hope there's an evaluation of this option and its advantages/disadvantages. In particular from a UX point of view.

> because policy matching is convoluted where the top level 
MeshHTTPRoute gets converted to a fake `to`.

this isn't convincing to me. The _main_ interface here is the policy structure, what does it matter that it gets converted to a fake `to`? Why don't we just change the implementation?

Reply:
There's a very strong intuition I have that it is _much_ clearer what is happening for a user when they simply see _one_ `spec.targetRef`, at the top level, pointing to a route. The policy applies to this route, done. No noise, no ambiguity.

As compared to when there's `spec.targetRef`, there's the policy namespace, there's multiple `spec.to[].targetRef`. How do these things interact?

Reply:
I've updated the MADR with more context and concrete advantages/disadvantages. If'you have any more questions, talking points please ask